js/src/common/helpers/listItems.js

```diff
@@ -30,21 +30,29 @@ export default function listItems(items) {
   if (!(items instanceof Array)) items = [items];
 
   return withoutUnnecessarySeparators(items).map((item) => {
-    const isListItem = item.component && item.component.isListItem;
-    const active = item.component && item.component.isActive && item.component.isActive(item.props);
-    const className = item.props ? item.props.itemClassName : item.itemClassName;
+    const isListItem = item.tag && item.tag.isListItem;
+    const active = item.tag && item.tag.isActive && item.tag.isActive(item.attrs);
+    const className = (item.attrs && item.attrs.itemClassName) || item.itemClassName;
 
     if (isListItem) {
       item.attrs = item.attrs || {};
       item.attrs.key = item.attrs.key || item.itemName;
+      item.key = item.attrs.key;
```

Comment:
Do we need `item.attrs.key` or `item.key`? Probably not both, right?

Reply:
Iirs, we're modifying it after it's been created. So if we want to change the key, we should change both the actual key and the attrs (for consistency)

Reply:
To elaborate, from my understanding of mithril, if `vnode.key` is set, `vnode.attrs.key` should also be set.